File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool Comp(vector<int> a, vector<int> b) {
    return a[1] < b[1];
}

int solution(vector<vector<int>> targets) {
    int answer = 0;
    
    sort(targets.begin(), targets.end(), Comp);
    
    int cur = 0;
    for(auto target: targets) {
        if(cur <= target[0]) {
            cur = target[1];
            answer++;
        }
    }
    
    return answer;
}

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.js
function solution(targets) {
    let answer = 0;
    
    targets.sort((a, b) => a[1] - b[1]);
    
    let cur = 0;
    for(const target of targets) {
        if(cur <= target[0]) {
            cur = target[1];
            answer++;
        }
    }
    
    return answer;
}